static/js/popUp.js

```diff
@@ -0,0 +1,28 @@
+import projects from './projectsList.js';
+
+export default function showPopUp(id) {
+  const popUpSection = document.createElement('section');
+  popUpSection.id = 'fullScreen-popUp';
+  const fullScreenCard = document.createElement('div');
+  fullScreenCard.className = 'fullScreen-card';
+  fullScreenCard.innerHTML += `
+    <div id="closePopUp">
+      <button type="button" class="button-clear"><img src="./static/images/Buttons/close.png" alt="close menu image"></button>
+    </div>
+    <h2 class="title">${projects[id].title}</h2>
+    <img class="fullScreenPopUp-image" src=${projects[id].image} alt="${projects[id].title1} image"/>
+    <ul class="tags">
+      ${
+  projects[id].tags.map((tag) => (`<li class="tag">${tag}</li>`)).join('')
+}
+    </ul>
+    <p>${projects[id].description}</p>
+    <div class="button-container">
+      <button class="button-primary" type="button">See Live <img src="./static/images/Buttons/seeLive.png" alt="live version image"></button>
+      <button class="button-primary" type="button">See Source <img src="./static/images/Buttons/github.png" alt="github image"> </button>
+    </div>
+  `;
+  popUpSection.appendChild(fullScreenCard);
+  document.getElementById('portfolio-popUP').appendChild(popUpSection);
+  document.querySelector('#closePopUp').addEventListener('click', () => popUpSection.remove());
+}
```

Comment:
**Optional Changes**
- [ ] There is a lot of HTML5 code here that could dynamically be generated using JavaScript on the fly. It always your code to follow the `SOLID` principle.

Reply:
Could you please explain it a little bit because I am not following the hint but I think it is a good one that I should work on